contracts/strategies/PlatypusVoterProxy.sol

```diff
@@ -0,0 +1,255 @@
+                address(this),
+                type(uint256).max
+            )
+        );
+        uint256 amount = toUint256(result, 0);
+        IERC20(_token).safeTransfer(msg.sender, amount);
+        return amount;
+    }
+
+    function emergencyWithdraw(
+        uint256 _pid,
+        address _stakingContract,
+        address _pool,
+        address _token,
+        address _asset
+    ) external onlyStrategy(_asset) {
+        platypusVoter.safeExecute(_stakingContract, 0, abi.encodeWithSignature("emergencyWithdraw(uint256)", _pid));
+        uint256 balance = IERC20(_asset).balanceOf(address(platypusVoter));
+        (uint256 expectedAmount, , ) = IPlatypusPool(_pool).quotePotentialWithdraw(_token, balance);
+        platypusVoter.safeExecute(_asset, 0, abi.encodeWithSignature("approve(address,uint256)", _pool, balance));
+        platypusVoter.safeExecute(
+            _pool,
+            0,
+            abi.encodeWithSignature(
+                "withdraw(address,uint256,uint256,address,uint256)",
+                _token,
+                balance,
+                expectedAmount,
+                msg.sender,
+                type(uint256).max
+            )
+        );
+        platypusVoter.safeExecute(_asset, 0, abi.encodeWithSignature("approve(address,uint256)", _stakingContract, 0));
+        platypusVoter.safeExecute(_token, 0, abi.encodeWithSignature("approve(address,uint256)", _pool, 0));
+    }
+
+    function claimReward(
+        address _stakingContract,
+        uint256 _pid,
+        address _asset
+    ) external onlyStrategy(_asset) {
+        (uint256 pendingPtp, address bonusTokenAddress, , uint256 pendingBonusToken) = IMasterPlatypus(_stakingContract)
+            .pendingTokens(_pid, address(platypusVoter));
+
+        uint256[] memory pids = new uint256[](1);
+        pids[0] = _pid;
+
+        platypusVoter.safeExecute(_stakingContract, 0, abi.encodeWithSignature("multiClaim(uint256[])", pids));
+
+        uint256 boostFee = 0;
+        if (boosterFee > 0 && boosterFeeReceiver > address(0) && platypusVoter.depositsEnabled()) {
+            boostFee = pendingPtp.mul(boosterFee).div(BIPS_DIVISOR);
+            platypusVoter.depositFromBalance(boostFee);
+            IERC20(address(platypusVoter)).safeTransfer(boosterFeeReceiver, boostFee);
+        }
+
+        uint256 stakingFee = 0;
+        if (stakerFee > 0 && stakerFeeReceiver > address(0)) {
+            stakingFee = pendingPtp.mul(stakerFee).div(BIPS_DIVISOR);
+            platypusVoter.safeExecute(
+                PTP,
+                0,
+                abi.encodeWithSignature("transfer(address,uint256)", stakerFeeReceiver, stakingFee)
+            );
+        }
+
+        uint256 reward = pendingPtp.sub(boostFee).sub(stakingFee);
+
+        platypusVoter.safeExecute(PTP, 0, abi.encodeWithSignature("transfer(address,uint256)", msg.sender, reward));
+
+        if (bonusTokenAddress > address(0)) {
+            platypusVoter.wrapAvaxBalance();
```

Comment:
is this supposed to handle AVAX/WAVAX as a bonus reward? if so, should it be outside of this clause?

Reply:
This handles all bonus rewards. It assumes that bonusTokenAddress will be set to WAVAX for WAVAX or AVAX rewards. We could add an extra check if bonusTokenAddress == WAVAX before wrapping AVAX balance.